compiler/noirc_frontend/src/parser/errors.rs

```diff
@@ -258,46 +270,46 @@ impl<'a> From<&'a ParserError> for Diagnostic {
                             .collect::<Vec<_>>()
                             .join(", ")
                     ),
-                    error.span,
+                    error.span(),
                 ),
                 ParserErrorReason::ExperimentalFeature(_) => {
-                    Diagnostic::simple_warning(reason.to_string(), "".into(), error.span)
+                    Diagnostic::simple_warning(reason.to_string(), "".into(), error.span())
```

Comment:
Would it make sense for `Diagnostic::simple_warning` to directly take a `Location`, to reduce the amount of `Location -> Span` conversions?

Reply:
I can try it. I tried to make the minimal changes in this PR to get it working, mainly because it's already huge. Let me know if you'd prefer me  to do it in this PR or a follow-up.